### PR TITLE
Pass connection string as argument

### PR DESCRIPTION
**Pull request by LukaszMoskala**
Hi,

I implemented passing connection string as second argument. If not specified, defaults to "/dev/ttyACM0,b115200".

I hope that this may be usefull for someone.

### CODE REVIEW

File: main.cc [hunk 1]
```diff
@@ -25,17 +25,25 @@ static bool reliable_write(int fd, const char *buffer, int len) {
...
-        fprintf(stderr, "usage: %s <gcode-file>\n", argv[0]);
+    if (argc < 2) {
+        fprintf(stderr, "usage: %s <gcode-file> [connection string]\n", argv[0]);
+        fprintf(stderr, "Example: %s file.gcode /dev/ttyACM0,b115200\n", argv[0]);
+
         return 1;
     }
     const char *filename = argv[1];
+
+    std::string connection_string="/dev/ttyACM0,b115200";
```
Comment (hzeller) [inline main.cc:40]: Given that the alternative string we get from argv[2] which alive when we call OpenMachineConnection(), we don't need to copy it into a std::string; so we can just assign it to a `const char *`. That way, we also don't have to do the c_str() conversion below in the OpenMachineConnection() and fprintf() call.

```
const char *connection_string = "/dev/ttyACM0,b115200";
```
Also please add a space between `if` and the open parenthesis to match the rest of the style in the file. Also, spaces around the assignment operator.

File: main.cc [hunk 2]
```diff
@@ -9,6 +9,10 @@
 #include <iostream>
 #include <string>
 
+using std::cerr;
```
Comment (hzeller) [inline main.cc:12]: I prefer the `fprintf(stderr, ...)` version

**Comment by LukaszMoskala** [comment]
Done. To be honest, I was not expecting that commits made after creating pull request will be added here.

**Comment by hzeller** [comment]
Thank you, merged!

**Comment by LukaszMoskala** [comment]
You're welcome :)

**Comment by hzeller** [comment]
In a follow-up commit, I've now also updated the usage output string and README to include the syntax of the connection string now that users  can change it.

**Pull request merged**